processor/logdedupprocessor/processor_test.go

```diff
@@ -221,3 +216,132 @@ func Test_unsetLogsAreExportedOnShutdown(t *testing.T) {
+	require.NoError(t, err)
+
+	// Consume the payload
+	err = p.ConsumeLogs(context.Background(), logs)
+	require.NoError(t, err)
+
+	// Wait for the logs to be emitted
+	require.Eventually(t, func() bool {
+		return logsSink.LogRecordCount() > 4
+	}, 3*time.Second, 200*time.Millisecond)
+
+	allSinkLogs := logsSink.AllLogs()
+	require.Len(t, allSinkLogs, 2)
+
+	expectedConsumedLogs, err := golden.ReadLogs("testdata/expected/conditionConsumedLogs.yaml")
+	require.NoError(t, err)
+	expectedDedupedLogs, err := golden.ReadLogs("testdata/expected/conditionDedupedLogs.yaml")
+	require.NoError(t, err)
+
+	consumedLogs := allSinkLogs[0]
+	dedupedLogs := allSinkLogs[1]
+
+	removeTimestamps(expectedConsumedLogs, expectedDedupedLogs, consumedLogs, dedupedLogs)
+
+	require.NoError(t, plogtest.CompareLogs(expectedConsumedLogs, consumedLogs, plogtest.IgnoreObservedTimestamp()))
+	require.NoError(t, plogtest.CompareLogs(expectedDedupedLogs, dedupedLogs, plogtest.IgnoreObservedTimestamp()))
+
+	// Cleanup
+	err = p.Shutdown(context.Background())
+	require.NoError(t, err)
+}
+
+func TestProcessorConsumeMultipleConditions(t *testing.T) {
+	logsSink := &consumertest.LogsSink{}
+	cfg := &Config{
+		LogCountAttribute: defaultLogCountAttribute,
+		Interval:          1 * time.Second,
+		Timezone:          defaultTimezone,
+		Conditions:        []string{`attributes["ID"] == 1`, `attributes["ID"] == 3`},
+		ExcludeFields: []string{
+			fmt.Sprintf("%s.remove_me", attributeField),
+		},
+	}
+
+	// Create a processor
+	p, err := newProcessor(cfg, logsSink, processortest.NewNopSettings())
+	p.conditions = getConditions(t, cfg.Conditions)
+	require.NoError(t, err)
+
+	err = p.Start(context.Background(), componenttest.NewNopHost())
+	require.NoError(t, err)
+
+	logs, err := golden.ReadLogs("testdata/input/conditionLogs.yaml")
+	require.NoError(t, err)
+
+	// Consume the payload
+	err = p.ConsumeLogs(context.Background(), logs)
+	require.NoError(t, err)
+
+	// Wait for the logs to be emitted
+	require.Eventually(t, func() bool {
+		return logsSink.LogRecordCount() > 3
+	}, 3*time.Second, 200*time.Millisecond)
+
+	allSinkLogs := logsSink.AllLogs()
+	require.Len(t, allSinkLogs, 2)
+
+	consumedLogs := allSinkLogs[0]
+	dedupedLogs := allSinkLogs[1]
+
+	expectedConsumedLogs, err := golden.ReadLogs("testdata/expected/multipleConditionsConsumedLogs.yaml")
+	require.NoError(t, err)
+	expectedDedupedLogs, err := golden.ReadLogs("testdata/expected/multipleConditionsDedupedLogs.yaml")
+	require.NoError(t, err)
+
+	removeTimestamps(expectedConsumedLogs, expectedDedupedLogs, consumedLogs, dedupedLogs)
+
+	require.NoError(t, plogtest.CompareLogs(expectedConsumedLogs, consumedLogs, plogtest.IgnoreObservedTimestamp()))
+	require.NoError(t, plogtest.CompareLogs(expectedDedupedLogs, dedupedLogs, plogtest.IgnoreObservedTimestamp()))
+
+	// Cleanup
+	err = p.Shutdown(context.Background())
+	require.NoError(t, err)
+}
+
+func removeTimestamps(plogs ...plog.Logs) {
```

Comment:
There should be an option in plogtest to ignore these attributes _values_. This still allows validation that the attributes are present.

Reply:
There wasn't an existing option to ignore log record attribute values, only one for ignoring resource attributes. I added one for ignoring log record attribute values and it is working great.
Let me know if adding that option is not something I should be doing in this PR.